src/video/video_driver.cpp

```diff
@@ -56,6 +56,20 @@ void VideoDriver::GameThread()
 	}
 }
 
+void VideoDriver::GameLoopPause()
+{
+	/* If we are not called from the game-thread, ignore this request. */
+	if (std::this_thread::get_id() != this->game_thread.get_id()) return;
+
+	this->game_state_mutex.unlock();
+
+	{
+		std::lock_guard<std::mutex> lock(this->game_thread_wait_mutex);
+	}
+
+	this->game_state_mutex.lock();
+}
```

Comment:
unlocking, then locking & immediately unlocking, then locking again without doing anything else inbetween looks really  weird :)

Reply:
That is what you get if OSes decide to optimize mutexes, and don't give other locks priority when you unlock :P Really stupid. Added a comment about it.